system/src/Grav/Common/Page/Collection.php

```diff
@@ -145,6 +145,18 @@ public function intersect(PageCollectionInterface $collection)
         return $this;
     }
 
+    /**
+     * Set current page.
+     */
+    public function setCurrent(string $path): void
+    {
+        reset($this->items);
+
+        while (key($this->items) !== $path && key($this->items) !== null) {
```

Comment:
I would get the key into a variable to speed up the loop.

Reply:
But key setting will have to be inside the loop anyway 🤔 Will see what I can come up with. Also will check how expensive this is

Reply:
Is it worth introducing a variable to gain 250ms in a 10 mil item collection? 🙂 Ran a few times and one time result was opposite even 🙂 
![image](https://user-images.githubusercontent.com/3602819/124972442-082c3180-e033-11eb-9a31-c528019d5ab6.png)

Reply:
It's 9% faster. :) Generally I prefer well-written code over repeating the same thing over and over again.

```
while (($key = key($this->items) !== $path) && $key !== null) {
```